app/vite.config.ts

```diff
@@ -1,25 +1,44 @@
 import { fileURLToPath, URL } from 'node:url'
 
-import { defineConfig } from 'vite'
+import { defineConfig, loadEnv } from 'vite'
 import vue from '@vitejs/plugin-vue'
 import inject from '@rollup/plugin-inject'
+import { ENV_LIST } from './env-list'
+import { isAddress } from 'ethers'
 
 // https://vitejs.dev/config/
-export default defineConfig({
-  plugins: [
-    vue(),
-    inject({
-      Buffer: ['buffer', 'Buffer']
+export default defineConfig(({ mode }) => {
+  // Validate .env file
+  const env = loadEnv(mode, process.cwd(), '')
+
+  if (!process.env.CI) {
+    ENV_LIST.forEach((key) => {
+      if (!env[key]) {
+        throw new Error(`Missing ${key} in .env file`)
+      } else if (key.includes('ADDRESS') && !isAddress(env[key])) {
+        throw new Error(`Invalid address in ${key}`)
+      } else if (key.includes('URL') && !env[key].startsWith('http')) {
+        throw new Error(`Invalid URL in ${key}`)
+      }
```

Comment:
i've also remark that if you extract the function here you can test it. 
Because what assure us is the test. 

With the test, we can test all different use case and be sure that when we provide this env we have this result.

Reply:
im sorry but i didnt get what u mean